File: ruby/src/common/SoupBuffet.rb
# SoupBuffet- A helper class

require "common/ChickenSoup"
require "common/ClamChowder"
require "common/FishChowder"
require "common/Minnestrone"
require "common/PastaFazul"
require "common/TofuSoup"
require "common/VegetableSoup"

class SoupBuffet

	
	attr_reader :soupBuffetName, :chickenSoup, :clamChowder, :fishChowder, :minnestrone, :pastaFazul, :tofuSoup, :vegetableSoup
	attr_writer :soupBuffetName, :chickenSoup, :clamChowder, :fishChowder, :minnestrone, :pastaFazul, :tofuSoup, :vegetableSoup

	def initialize()
		@soupBuffetName = nil
		@chickenSoup = nil
		@clamChowder = nil
		@fishChowder = nil
		@minnestrone = nil
		@pastaFazul = nil
		@tofuSoup = nil
		@vegetableSoup = nil
	end

	def setSoupBuffetName(soupBuffetNameIn)
		@soupBuffetName = soupBuffetNameIn
	end

	def setChickenSoup(chickenSoupIn) 
		@chickenSoup = chickenSoupIn
	end

	def setClamChowder(clamChowderIn)
		@clamChowder = clamChowderIn
	end

	def setFishChowder(fishChowderIn) 
		@fishChowder = fishChowderIn
	end

	def setMinnestrone(minnestroneIn)
		@minnestrone = minnestroneIn
	end

	def setPastaFazul(pastaFazulIn)
		@pastaFazul = pastaFazulIn
	end

	def setTofuSoup(tofuSoupIn)
		@tofuSoup = tofuSoupIn
	end

	def setVegetableSoup(vegetableSoupIn) 		
		@vegetableSoup = vegetableSoupIn
	end

	def getSoupBuffetName() 		
		@soupBuffetName
	end

	def getTodaysSoups() 		
		[
			" Today's Soups!  Chicken Soup: #{@chickenSoup.soupName}",
			"Clam Chowder: #{ @clamChowder.soupName }",
			"Fish Chowder: #{ @fishChowder.soupName }",
			"Minnestrone: #{ @minnestrone.soupName }",
			"Pasta Fazul: #{ @pastaFazul.soupName }",
			"Tofu Soup: #{ @tofuSoup.soupName }",
			"Vegetable Soup: #{ @vegetableSoup.soupName }",
		].join(', ')
	end
end

File: csharp/src/common/SoupBuffet.cs
//SoupBuffet.java - A helper class

namespace common {

using System.Collections.Generic;

public class SoupBuffet {

   public string soupBuffetName;
   
   public ChickenSoup chickenSoup;
   public ClamChowder clamChowder;
   public FishChowder fishChowder;
   public Minnestrone minnestrone;
   public PastaFazul pastaFazul;
   public TofuSoup tofuSoup;
   public VegetableSoup vegetableSoup;

   public string getSoupBuffetName() {
       return soupBuffetName;
   }
   public void setSoupBuffetName(string soupBuffetNameIn) {
      soupBuffetName = soupBuffetNameIn;
   }
   
   public void setChickenSoup(ChickenSoup chickenSoupIn) {
       chickenSoup = chickenSoupIn;
   }
   public void setClamChowder(ClamChowder clamChowderIn) {
       clamChowder = clamChowderIn;
   }
   public void setFishChowder(FishChowder fishChowderIn) {
      fishChowder = fishChowderIn;
   }
   public void setMinnestrone(Minnestrone minnestroneIn) {
      minnestrone = minnestroneIn;
   }
   public void setPastaFazul(PastaFazul pastaFazulIn) {
       pastaFazul = pastaFazulIn;
   }
   public void setTofuSoup(TofuSoup tofuSoupIn) {
       tofuSoup = tofuSoupIn;
   }
   public void setVegetableSoup(VegetableSoup vegetableSoupIn) {
       vegetableSoup = vegetableSoupIn;
   }
   
   public string getTodaysSoups() {
        List<string> stringOfSoups = new List<string>();
        stringOfSoups.Add(" Today's Soups!  ");
        stringOfSoups.Add(" Chicken Soup: ");        
        stringOfSoups.Add(this.chickenSoup.getSoupName()); 
        stringOfSoups.Add(" Clam Chowder: ");        
        stringOfSoups.Add(this.clamChowder.getSoupName()); 
        stringOfSoups.Add(" Fish Chowder: ");        
        stringOfSoups.Add(this.fishChowder.getSoupName()); 
        stringOfSoups.Add(" Minnestrone: ");        
        stringOfSoups.Add(this.minnestrone.getSoupName());
        stringOfSoups.Add(" Pasta Fazul: ");        
        stringOfSoups.Add(this.pastaFazul.getSoupName());
        stringOfSoups.Add(" Tofu Soup: ");        
        stringOfSoups.Add(this.tofuSoup.getSoupName());
        stringOfSoups.Add(" Vegetable Soup: ");        
        stringOfSoups.Add(this.vegetableSoup.getSoupName());
        return string.Join(", ", stringOfSoups.ToArray());          
   }
}

}        
